clang/lib/Headers/hlsl/hlsl_compat_overloads.h

```diff
@@ -54,5 +54,55 @@ clamp(U p0, V p1, W p2) {
   return clamp(p0, (U)p1, (U)p2);
 }
 
+//===----------------------------------------------------------------------===//
+// max builtin overloads
+//===----------------------------------------------------------------------===//
+
+template <typename T, typename U, uint N>
+constexpr __detail::enable_if_t<
+    __detail::is_arithmetic<U>::Value && (N > 1 && N <= 4), vector<T, N>>
+max(vector<T, N> p0, U p1) {
+  return max(p0, (vector<T, N>)p1);
+}
+
+template <typename T, typename U, uint N>
+constexpr __detail::enable_if_t<
+    __detail::is_arithmetic<U>::Value && (N > 1 && N <= 4), vector<T, N>>
+max(U p0, vector<T, N> p1) {
+  return max((vector<T, N>)p0, p1);
+}
+
+template <typename U, typename V>
+constexpr __detail::enable_if_t<
+    __detail::is_arithmetic<U>::Value && __detail::is_arithmetic<V>::Value, U>
+max(U p0, V p1) {
```

Comment:
This one is also problematic. DXC will not convert the second argument to the first type every time, it will prefer an int->float conversion over a float->int.